enacts/enactstozarr.py

```diff
@@ -11,34 +11,35 @@
 
 CONFIG = pingrid.load_config(os.environ["CONFIG"])
 
-RR_MRG_NC_PATH = CONFIG["onset"]["rr_mrg_nc_path"]
-RR_MRG_ZARR_PATH = CONFIG["onset"]["rr_mrg_zarr_path"]
-RESOLUTION = CONFIG["onset"]["rr_mrg_resolution"]
-CHUNKS = CONFIG["onset"]["rr_mrg_chunks"]
-
-#Read daily files of daily rainfall data
-#Concatenate them against added time dim made up from filenames
-
-RR_MRG_PATH = Path(RR_MRG_NC_PATH)
-RR_MRG_FILE = list(sorted(RR_MRG_PATH.glob("*.nc")))
+RESOLUTION = CONFIG["resolution"]
 
 def set_up_dims(xda):
+    
     datestr = Path(xda.encoding["source"]).name.split("_")[2]
     xda = xda.expand_dims(T = [dt.datetime(
       int(datestr[0:4]),
       int(datestr[4:6]),
       int(datestr[6:8])
     )])
     xda = xda.rename({'Lon': 'X','Lat': 'Y'})
+    
     return xda
 
-rr_mrg = xr.open_mfdataset(
-    RR_MRG_FILE,
-    preprocess = set_up_dims,
-    parallel=False
-).precip
-
-if not np.isclose(rr_mrg['X'][1] - rr_mrg['X'][0], RESOLUTION):
+def convert(variable):
+    print(f"converting files for: {variable}")
+    var_name = CONFIG['data_src'][variable][2]    
+    
+    nc_path = f"{CONFIG['nc_path']}{CONFIG['data_src'][variable][0]}"
+    
+    netcdf = list(sorted(Path(nc_path).glob("*.nc")))
+    
+    data = xr.open_mfdataset(
+        netcdf,
+        preprocess = set_up_dims,
+        parallel=False
+    )[var_name]
+    
+    if not np.isclose(data['X'][1] - data['X'][0], RESOLUTION):
     # TODO this method of regridding is inaccurate because it pretends
```

Comment:
@remicousin not sure if I was to keep this regridding code in the function since you had not originally included in the function you started working on. I put it in there for now. Mybe we can make it a parameter that can be defined so you can choose if you want to use the regridding or not?

Reply:
Sure, making `resolution` a parameter of the `convert` function makes sense.

Reply:
Just to confirm, it would be a parameter where if it is set to true then it will compare the resolution such as in line 42, to attempt the regridding?

Reply:
I imagined it would be a number. It could be optional--if it's not provided, then don't regrid. If it is provided, then check the resolution of the original dataset, and if it's not the same as the desired resolution, then regrid.

Reply:
So then do you have in mind a different test than line 42?

`if not np.isclose(data['X'][1] - data['X'][0], ZARR_RESOLUTION):`

`ZARR_RESOLUTION` would be the number it would check against. So then if we are to use `ZARR_RESOLUTION` as the parameter I would think we would need to 1) probably rename that var because it is vague that this is what it is being used for and 2) be more specific as to what you would set `ZARR_RESOLUTION` as in the config file if you do not want to test for regridding.

Reply:
The name of the parameter can just be `resolution`, and the configuration variable would be `CONFIG['onset']['resolution']`. Set it to `null` if you want to use the dataset's native resolution.